workers/deployment_manager/DeploymentManager/DeploymentService/DeploymentModifier.cs

```diff
@@ -0,0 +1,210 @@
+            }
+
+            var uploadSnapshotResponse =
+                snapshotServiceClient.UploadSnapshot(new UploadSnapshotRequest { Snapshot = snapshotToUpload });
+            snapshotToUpload = uploadSnapshotResponse.Snapshot;
+
+            // Upload content.
+            var httpRequest = WebRequest.Create(uploadSnapshotResponse.UploadUrl) as HttpWebRequest;
+            httpRequest.Method = "PUT";
+            httpRequest.ContentLength = snapshotToUpload.Size;
+            httpRequest.Headers.Set("Content-MD5", snapshotToUpload.Checksum);
+
+            using (var dataStream = httpRequest.GetRequestStream())
+            {
+                dataStream.Write(bytes, 0, bytes.Length);
+            }
+
+            // Block until we have a response.
+            httpRequest.GetResponse();
+
+            // Confirm that the snapshot was uploaded successfully.
+            var confirmUploadResponse = snapshotServiceClient.ConfirmUpload(new ConfirmUploadRequest
+            {
+                DeploymentName = snapshotToUpload.DeploymentName,
+                Id = snapshotToUpload.Id,
+                ProjectName = snapshotToUpload.ProjectName
+            });
+
+            return confirmUploadResponse.Snapshot.Id;
+        }
+
+        public static Deployment GetDeployment(string deploymentId, string projectName)
+        {
+            var request = new GetDeploymentRequest
+            {
+                Id = deploymentId,
+                ProjectName = projectName,
+            };
+
+            var response = deploymentServiceClient.GetDeployment(request);
+            return response.Deployment;
+        }
+
+        public static void AddDeploymentTag(string deploymentId, string projectName, string tag)
+        {
+            var deployment = GetDeployment(deploymentId, projectName);
+
+            if (!deployment.Tag.Contains(tag))
+            {
+                deployment.Tag.Add(tag);
+            }
+
+            var request = new UpdateDeploymentRequest
+            {
+                Deployment = deployment
+            };
+
+            deploymentServiceClient.UpdateDeployment(request);
+        }
+
+        public static void RemoveDeploymentTag(string deploymentId, string projectName, string tag)
+        {
+            var deployment = GetDeployment(deploymentId, projectName);
+
+            if (deployment.Tag.Contains(tag))
+            {
+                deployment.Tag.Remove(tag);
+            }
+
+            var request = new UpdateDeploymentRequest
+            {
+                Deployment = deployment
+            };
+
+            deploymentServiceClient.UpdateDeployment(request);
+        }
+
+        public static void UpdateDeploymentTag(string deploymentId, string projectName, string tagPrefix, string value)
+        {
+            var deployment = GetDeployment(deploymentId, projectName);
+            var tags = deployment.Tag.ToList();
+            foreach (var tag in tags)
```

Comment:
Wouldn't it be better to collapse the updates down into a single call to `UpdateDeployment`?